src/pygeosx/unitTests/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+#
+# Specify list of tests
+#
+set( pythonFiles
+  testWaveEquationAdjoint.py
+  )
+
+#
+# Specify list of xml files
+#
+set( xmlFiles
+  twoLayers2000x2000x2000.xml
+  )
+
+set( dependencyList pygeosx )
+
+if( GEOSX_BUILD_SHARED_LIBS )
+  set( dependencyList ${dependencyList} geosx_core )
+else()
+  set( dependencyList ${dependencyList} ${geosx_core_libs} )
+endif()
+
+if( ENABLE_CUDA )
+  set( dependencyList ${dependencyList} cuda )
+endif()
+
+# This macro parse a test file and extract test_ function from it to pass it to ctest
+macro( parse_test_file filename )
+  set( filename ${filename} )
+  file( READ "${filename}" contents )
+  string( REPLACE "\n" ";" rows ${contents} )
+  foreach( line ${rows} )
+    string( REGEX MATCHALL "^ *def *test_([A-Za-z_0-9])+ *" found ${line} )
+    if( found )
+      string( REGEX REPLACE "^ *def *test_(([A-Za-z_0-9])+) *" "\\1" name "${found}" )
+      configure_file( ${CMAKE_CURRENT_SOURCE_DIR}/pytest_template.sh.in ${CMAKE_CURRENT_BINARY_DIR}/${name}.sh
+        FILE_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE )
+      blt_add_test(NAME ${name} COMMAND ${CMAKE_CURRENT_BINARY_DIR}/${name}.sh)
+    endif()
+  endforeach()
+endmacro(parse_test_file)
```

Comment:
Aren't you doing the job of `pytest`?
That does not sound like the most efficient idea IMO.
`pytest` is supposed to be searching for unit tests and manage them.

Reply:
I propose we transform this test into an integrated test, since the adjoint test that is being performed requires a relatively large model to be run efficiently. I will just add a method to the current wave propagation test instead.
But there should be a discussion on how we could run PyGEOSX integrated tests within the current ats framework -- any ideas @TotoGaz?

Reply:
I have no idea when it comes to ATS (except that we should at least update to the last revision 😝 ).
But there is something that clashes with classical python fwk (like `nose` or `pytest`) when it comes to case discovery, case parametrization...
So I do not know how to handle this.